### PR TITLE
Academies db connection/update tests

### PR DESCRIPTION
**Pull request by lookupdaily**
This change removes the use of WireMock for mocking dependencies in our UI and Accessibility tests. 

We are now running our tests against a fake version of the Academies database (#194), and in this update we have updated our tests to use a static version of this fake trust data (#200) to check that the correct information about a trust is displayed in the application. 

We have also improved our page object models, to take away the details of what a user is searching for, or what they expect to see from our spec files. 

This is related to [User Story 134572](https://dev.azure.com.mcas.ms/dfe-gov-uk/Academies-and-Free-Schools-SIP/_workitems/edit/134572?McasTsid=26110&McasCtx=4): Build: Connect to Academies DB

### Future improvements
Currently the developer will need to run the `FindInformationAcademiesTrust.Data.AcademiesDb.Faker` project and copy the `data/trusts.json` file to the playwright tests folder any time that the Faker project is updated, to ensure tests pass locally and in the pipeline. In the future we would like to automate this.

## Changes

- Remove WireMock configuration from test set up
- Simplify spec files to remove search terms
- Add FakeTestData class, which is responsible for parsing json data, and finding appropriate trust data to use in UI tests
- Add header spec which tests that different types of trust are displayed correctly in the trust header
- Add assertions for additional properties UID and GroupID shown in search results (#168 )
- Update documentation for running tests locally, and for connecting to the fake database for local development

## Screenshots of UI changes

N/A

## Checklist

- [ ] Deploy this branch to the test environment for manual testing once comments have been resolved and checks have passed 
- [ ] Attach this pull request to the appropriate user story in Azure DevOps
- [ ] Update the ADR decision log if needed

### CODE REVIEW

File: tests/playwright/page-object-model/search-page.ts [hunk 1]
```diff
@@ -27,25 +37,40 @@ export class SearchPage {
...
+    this.currentSearch.term = searchTerm
     await this.page.goto(`/search/?keywords=${searchTerm}`)
   }
 
+  async goToPageWithResults (): Promise<void> {
+    await this.goToSearchFor('mary')
+  }
+
+  async goToPageWithNoResults (): Promise<void> {
+    await this.goToSearchFor('non')
```
Comment (dynamictulip) [inline tests/playwright/page-object-model/search-page.ts:49]: We should make this a little longer and definitely invalid so there's no chance future test data will match it

File: tests/playwright/page-object-model/shared/search-form-component.ts [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
-  ukprn: string
-  academiesInTrustCount: number
+class SelectedTrust {
+  name = ''
+  address = ''
+}
+
+export class CurrentSearch {
+  term = ''
+  selectedTrust = new SelectedTrust()
```
Comment (dynamictulip) [inline tests/playwright/page-object-model/shared/search-form-component.ts:10]: If this is just for autocomplete, should we rename it to reflect that? So it doesn't get confused with trusts from fakedata

Comment (lookupdaily) [reply tests/playwright/page-object-model/shared/search-form-component.ts:10]: It is also for if a user clicks on a trust in the search results list

File: tests/playwright/page-object-model/trust/contacts-page.ts [hunk 3]
```diff
@@ -1,39 +1,49 @@
...
     this.trustNavigation = new TrustNavigationComponent(page)
     this.pageHeadingLocator = page.locator('h1')
   }
 
-  async goTo (
-    ukprn: string = MockTrustsProvider.expectedFormattedTrustResult.ukprn
-  ): Promise<void> {
-    await this.page.goto(`/trusts/contacts/${ukprn}`)
+  async goTo (): Promise<void> {
+    const uid = this.fakeTestData.getFirstTrust().uid
```
Comment (dynamictulip) [inline tests/playwright/page-object-model/trust/contacts-page.ts:23]: We could have a field in this class for the currently selected trust and set it here. We could then use it elsewhere in the class and assertions

Comment (dynamictulip) [reply tests/playwright/page-object-model/trust/contacts-page.ts:23]: This pattern should work for trust details too.

File: tests/playwright/playwright.ci.config.ts
Comment (dynamictulip) [inline tests/playwright/playwright.ci.config.ts:1]: Not for this PR......do we actually need this file anymore?

File: tests/playwright/page-object-model/trust/contacts-page.ts (hunk 3 above)
Comment (lookupdaily) [reply tests/playwright/page-object-model/trust/contacts-page.ts:23]: There is an issue with this for the navigation spec, as we are not navigating to the page using 'goTo' - should we leave as is, or check for an undefined 'currentTrust' in 'toBeOnTheRightPage'?

**Review by dynamictulip — APPROVED** [review]

File: tests/playwright/page-object-model/trust/contacts-page.ts (hunk 3 above)
Comment (dynamictulip) [reply tests/playwright/page-object-model/trust/contacts-page.ts:23]: `toBeOnTheRightPage` (for details and contacts) is only used by the navigation tests. I think we should remove the trust check

**Review by dynamictulip — APPROVED** [review]

**Review by nwarms — APPROVED** [review]

**Pull request merged**